services/ensquare.js

```diff
@@ -21,16 +21,18 @@ export function getB2File(path) {
 export async function getLocalizedData(data, locale = "en") {
   const jaData = await getData(data, "ja", true);
   const enFanData = await getData(data, "en");
-  const enData = await getData(data, "en", true);
 
-  let localized = [enFanData, jaData, enData];
+  const localized = { mainLang: {}, subLang: {} };
   if (locale === "ja") {
-    localized = [jaData, enFanData, enData];
+    localized.mainLang = jaData;
+    localized.subLang = enFanData;
+  } else {
+    localized.mainLang = enFanData;
+    localized.subLang = jaData;
   }
   return {
     main: jaData,
-    localized: localized.filter((l) => l.status === "success"),
-    localized_full: localized,
```

Comment:
The choice of array for `localized` was a bit sus to me after looking at how `getLocalizedData` was being used, but please lmk if I made any wrong assumptions and I can revert my changes!

I think the assumption that the first item in the array is the locale's language and the second is the sub-language is less readable to other devs, especially if you filter the array based on request status (which could change the order if one of the requests fails). Using an object with `mainLang` and `subLang` keys seems more intentional to me, although it depends on whether we want to handle more than one main language and sub language.

It also looked like `enData` wasn't being referenced so I removed that!

Reply:
hold on i just saw this, yeah i plan to add at least 3 languages (not all is shown immediately but etc on hover) so :toyathink:, altho i also do think having both numbered indices *and* main/sub lang would be nice (bc the names immediately shown are only 2 at most anyways)

Reply:
also, localized and localized full is intentional:
- some parts of the ui would only have 1-2 immediately shown props, and showing an error on one of two props can be kind of confusing, so in those cases i'd like to have it just remove any unsuccesful fetches
- on the other hand, on a screen where the user wants to see the *full* list (etc. if theyre purposefully clicking on the names of a card), itd be nice to show the user if a card's language is __unavailable__ and not hide that

as for why i dont do the filtering client side, `localized` is used a lot more often so having to filter every time feels weird. though, i am thinking of making some sort of function (or class but cluelesshiiro) so data doesnt have to be sent twice. the package size for the page is like 3MB rn which is kinda huge (i actually also wanna do a "property filter" thing so instead of getting *everything* getLocalizedData can just have an array of what it actually needs to fetch, etc. the cards list page doesnt need info on spp for example, to make the bundle size smaller

Reply:
feel free to suggest changes too bc its kind of a janky solution but i havent really come up with smth better, or at least i havent written them down in code bc i wanted to make it work first slkfjsdlk

Reply:
Ah ok, I think based on the current usage and context for future plans I agree it makes sense to add back `localized` and `localized_full`! 

It does seem like `localized_full` could get really big but the only solution I can think of is to break down `getLocalizedData` into separate requests to fetch exactly what's needed, like `getDataInLocale` and `getDataAllLocales`. The property filter idea works too!